### PR TITLE
feat(settings/admin/ai): Add Task Processing API settings

### PR DESCRIPTION
**Pull request by marcelklehr**
## Summary
Adds the settings UI for the TaskProcessing API

## Todo

- [ ] Discuss whether we want to have a **do not offer this task** option

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [ ] Screenshots before/after for front-end changes
- [ ] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [ ] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

File: lib/private/TaskProcessing/Manager.php [hunk 1]
```diff
@@ -497,11 +499,23 @@ public function getProviders(): array {
 	}
 
 	public function getPreferredProvider(string $taskType) {
-		$providers = $this->getProviders();
-		foreach ($providers as $provider) {
-			if ($provider->getTaskTypeId() === $taskType) {
-				return $provider;
+		try {
+			$preferences = json_decode($this->config->getAppValue('core', 'ai.taskprocessing_provider_preferences', 'null'));
```
Comment (kyteinsky) [inline lib/private/TaskProcessing/Manager.php:503]: [JSON_THROW_ON_ERROR](https://www.php.net/manual/en/json.constants.php#constant.json-throw-on-error) flag here

Comment (marcelklehr) [reply lib/private/TaskProcessing/Manager.php:503]: right you are, sir :see_no_evil:

File: apps/settings/src/components/AdminAI.vue [hunk 2]
```diff
@@ -4,6 +4,33 @@
 -->
 <template>
 	<div>
+		<NcSettingsSection :name="t('settings', 'Unified task processing')"
+			:description="t('settings', 'AI tasks can be implemented by different apps. Here you can set which app should be used for which task.')">
+			<template v-for="type in taskProcessingTaskTypes">
+				<div :key="type">
+					<h3>{{ t('settings', 'Task:') }} {{ type.name }}</h3>
```
Comment (kyteinsky) [inline apps/settings/src/components/AdminAI.vue:11]: image

h3 tag for the tasks seems to be a bit larger than the "Unified task processing" heading

Comment (marcelklehr) [reply apps/settings/src/components/AdminAI.vue:11]: Yeah, semantically it's correct, though I think. As the "Unified task processing" is h2, I can make the h3 smaller?

Comment (kyteinsky) [reply apps/settings/src/components/AdminAI.vue:11]: that would make sense

**Comment by marcelklehr** [comment]
image

**Review by kyteinsky — APPROVED** [review]
:rocket:

**Comment by bigcat88** [comment]
am I understand right that this page will list all available registered task types?

**Comment by marcelklehr** [comment]
> am I understand right that this page will list all available registered task types?

yes

**Review by bigcat88 — APPROVED** [review]

**Review by julien-nc — APPROVED** [review]
Works like a charm. Could you check if [this issue](https://github.com/nextcloud/server/pull/43993) was not reintroduced?

**Comment by julien-nc** [comment]
* Rebased on master
* Set min-width for the selects to 350px
* Compiled the assets

**Comment by marcelklehr** [comment]
> Could you check if https://github.com/nextcloud/server/pull/43993 was not reintroduced?

It was not afaict

**Comment by julien-nc** [comment]
@marcelklehr Can you check my fix in the last commit? https://github.com/nextcloud/server/pull/46509/commits/183726a299356e3ced6ad3b54ffe5f6d5273c896

**Comment by marcelklehr** [comment]
Looks good

**Pull request merged**